File: app/src/main/java/com/example/customviewprojdct/activity/ClassOneActivity.java
package com.example.customviewprojdct.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.FrameLayout;

import com.example.customviewprojdct.R;
import com.example.customviewprojdct.view.class1.DashboardView;
import com.example.customviewprojdct.view.class1.PieView;
import com.example.customviewprojdct.view.class1.TestView;

public class ClassOneActivity extends AppCompatActivity implements View.OnClickListener {

    public static void start(Context context) {
        context.startActivity(new Intent(context, ClassOneActivity.class));
    }

    FrameLayout fl_content;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_class_one);
        fl_content = findViewById(R.id.fl_content);
        findViewById(R.id.tvDash).setOnClickListener(this::onClick);
        findViewById(R.id.tvPie).setOnClickListener(this::onClick);
        findViewById(R.id.tvText).setOnClickListener(this::onClick);

    }

    @Override
    public void onClick(View v) {
        fl_content.removeAllViews();
        View targetView = null;
        switch (v.getId()) {
            case R.id.tvText:
                targetView = new TestView(this);
                break;
            case R.id.tvDash:
                targetView = new DashboardView(this);
                break;
            case R.id.tvPie:
                targetView = new PieView(this);
                break;
        }
        if (targetView != null) {
            fl_content.addView(targetView);
        }
    }
}

File: app/src/main/java/com/example/customviewprojdct/activity/ClassOneActivity.kt
package com.example.customviewprojdct.activity

import android.content.Context
import android.content.Intent
import android.view.View
import com.example.customviewprojdct.extensions.showToast
import com.example.customviewprojdct.view.class1.DashboardView
import com.example.customviewprojdct.view.class1.PieView
import com.example.customviewprojdct.view.class1.TestView
import com.example.customviewprojdct.view.class2.AvatarView
import com.example.customviewprojdct.view.class2.ScratchCardView
import com.example.customviewprojdct.view.class2.TextLoadingView
import com.example.customviewprojdct.view.class2.XfermodeViewDemoView

class ClassOneActivity : BaseClassActivity() {
    companion object {
        fun start(context: Context) {
            context.startActivity(Intent(context, ClassOneActivity::class.java))
        }
    }

    override fun initData(): MutableList<Triple<String, String, Class<out View>?>> {
        return mutableListOf(
            Triple("1", "测试", TestView::class.java),
            Triple("2", "仪表盘", DashboardView::class.java),
            Triple("3", "饼图", PieView::class.java)
        )
    }

}